client/src/store/modules/search/actions.js

```diff
@@ -173,146 +173,141 @@ async function searchGraphql(
-    return result;
-  };
-  const spotifyNodes = await searchSpotify(searchString, rootState, dispatch, [
-    "album",
-    "song",
-  ]);
+  // Gets nodetypes based on endpoint
+  // Only spotify
+  const spotifyNodeTypes = rootState.schema.nodeTypes
+    .filter((nodeType) => nodeType.endpoints.includes("spotify"))
+    .map((nodeType) => nodeType.label);
 
-  const graphQlNodeTypes = rootState.schema.nodeTypes.filter((nodeType) => {
-    return (
-      nodeType.endpoints.includes("graphql") &&
-      !nodeType.endpoints.includes("spotify")
-    );
-  });
+  //Only GraphQL
+  const graphQlNodeTypes = rootState.schema.nodeTypes.filter((nodeType) =>
+    nodeType.endpoints.includes("graphql"),
+  );
 
-  const graphQlAndSpotifyNodeTypes = rootState.schema.nodeTypes.filter(
-    (nodeType) => {
-      return (
+  //Spotify and graphql
+  const graphQlAndSpotifyNodeTypes = rootState.schema.nodeTypes
+    .filter(
+      (nodeType) =>
         nodeType.endpoints.includes("graphql") &&
-        nodeType.endpoints.includes("spotify")
-      );
-    },
+        nodeType.endpoints.includes("spotify"),
+    )
+    .map((nodeType) => nodeType.label);
+
+  //Searches for nodes only with spotify endpoint
+  const spotifyNodes = await searchSpotify(
+    searchString,
+    rootState,
+    dispatch,
+    spotifyNodeTypes,
   );
 
+  //Searches for nodes with that include graphql endpoint
   const graphQlNodes = await searchGraphql(
     [
       { attributeSearch: "name", operator: "=", attributeData: searchString },
-      { attributeSearch: "limit", operator: "=", attributeData: 50 },
+      { attributeSearch: "limit", operator: "=", attributeData: limit },
     ],
     graphQlNodeTypes,
     dispatch,
     rootState,
   );
 
-  const graphQlAndSpotifyNodes = await fetchFromGraphQlThenSpotify();
+  //If no nodes were found in graphql search for a given node type, and it has a spotify endpoint, use the spotify endpoint to check for entries
+  //Gets all found node types
+  const foundTypes = graphQlNodes.map((node) => node.data.label);
+  //Gets all node types that haven't been found
+  const additionalSpotifyNodeTypes = _.difference(
+    graphQlAndSpotifyNodeTypes,
+    foundTypes,
+  );
+  //Searches spotify if type was not found at all
+  const graphQlAndSpotifyNodes =
+    additionalSpotifyNodeTypes.length > 0
+      ? await searchSpotify(
+          searchString,
+          rootState,
+          dispatch,
+          graphQlAndSpotifyNodeTypes,
+        )
+      : [];
+
+  additionalSpotifyNodeTypes.forEach((type) =>
+    checkNodesExistence(
+      type,
+      graphQlAndSpotifyNodes.map((node) => node.data.sid),
+      rootState.schema,
+      dispatch,
+      rootState,
+    ),
+  );
 
   return [...spotifyNodes, ...graphQlNodes, ...graphQlAndSpotifyNodes];
 }
+
+const searchNodeTypeInGraphQl = async (
+  schemaNodeType,
+  searchString,
+  limit,
+  rootState,
+  dispatch,
+) => {
+  if (!schemaNodeType.endpoints.includes("graphql")) return [];
```

Comment:
Perhaps throw an error instead so the developer is alerted that this is not the intended use

Reply:
This wouldn't work, because the function is always invoked, regardless of the endpoint. This would lead to wrong errors when only having "graphql" or "spotify".